### PR TITLE
Added *fetch-through-merge-base

### PR DESCRIPTION
**Pull request by michal-dagan**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-8945)

## Description
- Added the [fetch_through_merge_base](https://github.com/rmacklin/fetch-through-merge-base/blob/main/fetch_through_merge_base.sh).
- Removed the call to Utils/release_notes_generator.py since it is no longer in use.
([failed on Master](https://code.pan.run/xsoar/content/-/pipelines/6928865) with "Unable to get the SHA1 of the commit in which the version was released.")

trigger SDK nightly: https://code.pan.run/xsoar/content/-/pipelines/6933252

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: .gitlab/ci/.gitlab-ci.on-push.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ trigger-private-build:
 .create-release-notes-and-common-docs:
   - section_start "Create Release Notes and Common Server Documentation" --collapsed
   - echo "Creating Release Notes and Content Descriptor"
-  - python3 Utils/release_notes_generator.py $CONTENT_VERSION $GIT_SHA1 $CI_BUILD_ID --packs-release-notes "${ARTIFACTS_FOLDER_SERVER_TYPE}/packs-release-notes.md" --content-descriptor "${ARTIFACTS_FOLDER_SERVER_TYPE}/content-descriptor.json" --github-token $GITHUB_TOKEN
+#  - python3 Utils/release_notes_generator.py $CONTENT_VERSION $GIT_SHA1 $CI_BUILD_ID --packs-release-notes "${ARTIFACTS_FOLDER_SERVER_TYPE}/packs-release-notes.md" --content-descriptor "${ARTIFACTS_FOLDER_SERVER_TYPE}/content-descriptor.json" --github-token $GITHUB_TOKEN
```
Comment (DeanArbel) [inline .gitlab/ci/.gitlab-ci.on-push.yml:42]: Why is this commented out?

**Review by DeanArbel — APPROVED** [review]
Looks good to me.

File: .gitlab/ci/.gitlab-ci.on-push.yml (hunk 1 above)
Comment (michal-dagan) [reply .gitlab/ci/.gitlab-ci.on-push.yml:42]: From what I understand, it is not being used and it uses a commit from two years ago.

Comment (ilaner) [reply .gitlab/ci/.gitlab-ci.on-push.yml:42]: Yes, we need to delete it and the comment
@DeanArbel We used to use it when we had content releases.

File: .gitlab/ci/.gitlab-ci.global.yml [hunk 2]
```diff
@@ -246,6 +246,12 @@
   - |
   - section_end "Checkout upload commit content nightly"
 
+.fetch-through-merge-base: &fetch-through-merge-base
+  - section_start "Fetch Through Merge Base"
+  - chmod +x ./Tests/scripts/fetch_through_merge_base.sh
```
Comment (ilaner) [inline .gitlab/ci/.gitlab-ci.global.yml:251]: please run this locally and commit, and you can delete this file after the commit

File: Tests/scripts/fetch_through_merge_base.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+DEEPEN_LENGTH=${DEEPEN_LENGTH:-10}
+
+HEAD_REF=${1:-$(git rev-parse HEAD)}
+BASE_REF=${2:-origin/master}
+
+while [ -z "$(git merge-base "$BASE_REF" "$HEAD_REF" 2>/dev/null)" ]; do
+  echo "Continuing fetch with a depth of $DEEPEN_LENGTH commits..."
+  git fetch -q --deepen="$DEEPEN_LENGTH" origin
```
Comment (michal-dagan) [inline Tests/scripts/fetch_through_merge_base.sh:10]:
```suggestion
  git fetch -q --deepen="$DEEPEN_LENGTH" origin "$BASE_REF" "$HEAD_REF";
```

File: .gitlab/ci/.gitlab-ci.on-push.yml [hunk 4]
```diff
@@ -39,6 +39,10 @@ trigger-private-build:
 .create-release-notes-and-common-docs:
   - section_start "Create Release Notes and Common Server Documentation" --collapsed
   - echo "Creating Release Notes and Content Descriptor"
+  - |
+    if [ "${CI_COMMIT_BRANCH}" != "master" ]; then
```
Comment (michal-dagan) [inline .gitlab/ci/.gitlab-ci.on-push.yml:43]:
```suggestion
    if [ "${CI_COMMIT_BRANCH}" == "master" ]; then
```

File: Tests/scripts/fetch_through_merge_base.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+DEEPEN_LENGTH=${DEEPEN_LENGTH:-$GIT_DEPTH}
+
+HEAD_REF=${1:-$(git rev-parse HEAD)}
+BASE_REF=${2:-origin/master}
+
+while [ -z "$(git merge-base "$BASE_REF" "$HEAD_REF" 2>/dev/null)" ]; do
+  echo "Continuing fetch with a depth of $DEEPEN_LENGTH commits..."
+  git fetch --deepen="$DEEPEN_LENGTH" origin "$BASE_REF" "$HEAD_REF";
```
Comment (michal-dagan) [inline Tests/scripts/fetch_through_merge_base.sh:10]:
```suggestion
  git fetch --deepen="$DEEPEN_LENGTH" origin;
```

[the remark this answers is not in the document]
Comment (michal-dagan) [reply Tests/scripts/fetch_through_merge_base.sh]: already done.

File: Tests/scripts/fetch_through_merge_base.sh [hunk 6]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+DEEPEN_LENGTH=${DEEPEN_LENGTH:-$GIT_DEPTH}
+
+HEAD_REF=${1:-$(git rev-parse HEAD)}
+BASE_REF=${2:-origin/master}
+
+while [ -z "$(git merge-base "$BASE_REF" "$HEAD_REF" 2>/dev/null)" ]; do
+  echo "Continuing fetch with a depth of $DEEPEN_LENGTH commits..."
+  git fetch --deepen="$DEEPEN_LENGTH" origin;
```
Comment (michal-dagan) [inline Tests/scripts/fetch_through_merge_base.sh:10]:
```suggestion
  git fetch -q --deepen="$DEEPEN_LENGTH" origin;
```

File: .gitlab/ci/.gitlab-ci.on-push.yml [hunk 7]
```diff
@@ -39,6 +39,10 @@ trigger-private-build:
 .create-release-notes-and-common-docs:
   - section_start "Create Release Notes and Common Server Documentation" --collapsed
   - echo "Creating Release Notes and Content Descriptor"
+  - |
+    if [ "${CI_COMMIT_BRANCH}" == "master" ]; then
+      git fetch --depth=0
```
Comment (michal-dagan) [inline .gitlab/ci/.gitlab-ci.on-push.yml:44]:
```suggestion
      git fetch -q --depth=0
```

**Review by kobymeir — APPROVED** [review]

File: .gitlab/ci/.gitlab-ci.global.yml [hunk 8]
```diff
@@ -246,6 +246,12 @@
   - |
   - section_end "Checkout upload commit content nightly"
 
+.fetch-through-merge-base: &fetch-through-merge-base
+  - section_start "Fetch Through Merge Base"
+  - ./Tests/scripts/fetch_through_merge_base.sh "$CI_COMMIT_SHA" "origin/master"
+  - sleep 10
+  - ./Tests/scripts/fetch_through_merge_base.sh "$CI_COMMIT_SHA" "$LAST_UPLOAD_COMMIT"
+  - section_end "Fetch Through Merge Base Completed"
```
Comment (tkatzir) [inline .gitlab/ci/.gitlab-ci.global.yml:254]:
```suggestion
  - section_end "Fetch Through Merge Base"
```

**Pull request closed without merging**